tutorials/migration.md

```diff
@@ -4,96 +4,846 @@ layout: intro
+
+Next install the openHAB 1.x compatibility layer using the Karaf Console 
+instructions above.
+
+Copy your openhab.cfg file to `<openHAB 2 conf>/services`. If you are running an apt-get 
+installed openHAB 1.x openhab.cfg is located in `/etc/openhab/configurations`.
+For an apt-get installed openHAB 2 the `<openHAB 2 conf>` folder is located in 
+`/etc/openhab2`. For manually installed openHAB 1.x this file is located
+in `<openhab home>/configurations` and in manually installed openHAB 2 it is
+`<openhab home>/conf`.
+
+This file will have a lot of redundant configuration information in it that you
+have already moved over to individual binding .cfgs. Make sure to comment those
+out and just leave the config parameters for the bindings you are manually 
+installing.
+
+Now copy the add-on's jar files that you want to install to the `<openHAB conf>/addons` folder for 
+openHAB 2. For example:
+
+```
+cp /usr/share/openhab/addons/org.openhab.binding.astro-1.8.3.jar /usr/share/openhab2/addons
+```
+
+There may be some errors about there not being a `bind` and `unbind` method. If
+everything worked you should see in the log the same entries from the binding 
+you see when openHAB 1.x starts.
+
+Watch the logs for errors as they can be informative. For example, some Actions
+require a corresponding Binding be installed first. An example error of this 
+sort looks like:
+
+```
+2016-09-08 15:15:04.613 [WARN ] [org.apache.felix.fileinstall        ] - Error while starting bundle: file:/openhab/addons/org.openhab.action.astro-1.8.3.jar
+org.osgi.framework.BundleException: Could not resolve module: org.openhab.action.astro [210]
+  Unresolved requirement: Import-Package: org.openhab.binding.astro.internal.calc
+```
+
+Note, the above error is for illustration purposes. You should not be 
+installing Astro this way.
+
+###Final Add-ons Installation Steps
+
+openHAB 2 has a different 
+[folder layout](http://docs.openhab.org/installation/linux.html#file-locations).
+Of particular node are the configuration folders and userdata folders. See the
+previous link for the location of these folders for your installtion.
+
+As with openHAB 1.x, one must restart openHAB 2 to pick up changes to .cfg files.
+Therefore restart openHAB 2 now.
+
+If you haven't already, configure default persistence now. You can do this 
+through `runtime.cfg` or PaperUI.
+
+Now openHAB 2 is ready to receive your other configurations, items and such. 
+Watch the log file as you transfer over files for errors. These errors will 
+have to be fixed.
+
+Copy over your files in the following order:
+
+~~~~
+cp <openHAB 1.x conf>/configurations/transform/* <openHAB 2 conf>/transform/*
+cp <openHAB 1.x conf>/configurations/scripts/* <openHAB 2 conf>/scripts/*
+cp <openHAB 1.x conf>/configurations/persistence/* <openHAB 2 conf>/persistence
+Copy any custom icons added from webapps/images to <openHAB 2 conf>/icons/classic
+Copy any custom webviews from webapps tp <openHAB 2 conf>/html
```

Comment:
Because they are more involved than simply copying everything in those folders over. For the icons one only wants to copy over the custom ones, not all of them.

Similarly there may be things in webviews (e.g. habmin) that one would not want to copy over to the OH 2 install.